src/Orleans.Core/Configuration/Options/ReminderOptions.cs

```diff
@@ -0,0 +1,51 @@
+
+using System;
+
+namespace Orleans.Hosting
+{
+    public class ReminderOptions
+    {
+        public static class BuiltIn
+        {
+            public const string NotSpecified = "NotSpecified";
+
+            /// <summary>Grain is used to store reminders information. 
+            /// This option is not reliable and thus should only be used in local development setting.</summary>
+            public const string ReminderTableGrain = "ReminderTableGrain";
+
+            /// <summary>AzureTable is used to store reminders information. 
+            /// This option can be used in production.</summary>
+            public const string AzureTable = "AzureTable";
+
+            /// <summary>SQL Server is used to store reminders information. 
+            /// This option can be used in production.</summary>
+            public const string SqlServer = "SqlServer";
+
+            /// <summary>Used for benchmarking; it simply delays for a specified delay during each operation.</summary>
+            public const string MockTable = "MockTable";
+
+            /// <summary>Reminder Service is disabled.</summary>
+            public const string Disabled = "Disabled";
+
+            /// <summary>Use custom Reminder Service from third-party assembly</summary>
+            public const string Custom = "Custom";
+        }
+
+        /// <summary>
+        /// The ReminderService attribute controls the type of the reminder service implementation used by silos.
+        /// </summary>
+        public string ReminderService { get; set; }
```

Comment:
same here, string vs Type

Reply:
This is a good example of why types in config can be a problem.   We support several reminder types, but we can't case of them an load the related assemblies by type, because if the assembly is not loaded we can't have the type.  Please take a look at ReminderTableFactory's usage of this value and consider how that logic can be performed with Type.

Reply:
sorry I wasn't clear on what my concerns are. The usage in `ReminderTableFactory` is the reason I raised this concern. Using assembly loading to create reminder table isn't compile time safe and more error prone. The pattern used in other places to avoid this, say GatewayListProvider, is we have extension methods like `UseAzureGatewayListProvider` to configure DI with the proper gateway list provider. This avoided using string to map to the related provider type, and use assembly loader to load the assembly and then create the provider. 

Whether this is out of the scope of this PR is debatable. Since for gatewaylist provider, creating those extension methods and refactor the infra is part of the work for move gatewatlist provider to use options. But if we decide it is out of scope, at least we add a todo to go back and refactor this.

Reply:
I agree that refactoring systems to fit the new model needs be done, and in some cases  (as with the strategies) I've done this, but in other cases I've not.  So minimally I've been inconsistent on this effort.  Given the size and complexity of this PR I suggest we push to future iterations any changes that we can.  As you've also touched on, we can (should) note these and revisit them in future passes (prior to 2.0 RC).

Reply:
agreed